src/main/java/net/starschema/clouddb/jdbc/Oauth2Bigquery.java

```diff
@@ -103,11 +103,11 @@
       HttpTransport httpTransport,
       String userAgent,
       String rootUrl,
-      String targetServiceAccount,
+      List<String> targetServiceAccounts,
       @Nullable String oauthToken) {
 
-    if (targetServiceAccount != null) {
-      credential = impersonateServiceAccount(credential, targetServiceAccount);
+    if (!targetServiceAccounts.isEmpty()) {
```

Comment:
On style nit is that you could move this check into `impersonateServiceAccount()`.  If it's empty, then you just return the passed in credential, otherwise you perform the impersonation logic.  The benefit there would be that you don't require as much from the caller of `impersonateServiceAccount()`, they just pass in a list, and the impersonate function deals with it being empty in a consistent way.  If we had multiple places calling the function, that would be more apparent, but with just the single call site, it really is just a nit.

Reply:
Sounds good, thanks for catching that. Currently, no checking occurs in `impersonateServiceAccount` which does feel wrong. Nothing to stop a caller from passing in an empty array and blowing things up.